File: Python/Algorithms/101.py
class Solution:
    def isSymmetric(self, root) -> bool:
        left, right = root.left, root.right
        if not left and not right:
            return True
        if any(i is None for i in [left, right]):
            return False

        # Use a stack for each subtree.
        sleft, sright = [left], [right]
        while sleft or sright:
            left, right = sleft.pop(), sright.pop()

            # Value check.
            if left.val != right.val:
                return False

            # grab children for each
            lleft, lright = left.left, left.right
            rleft, rright = right.left, right.right
            # and append them symmetrically.
            if lleft and rright:
                sleft.append(lleft)
                sright.append(rright)
            else:
                # One of them is None. If not both,
                # we got a mismatch.
                if lleft != rright:
                    return False
            if lright and rleft:
                sleft.append(lright)
                sright.append(rleft)
            else:
                if lright != rleft:
                    return False
        return sleft == sright

File: Javascript/Algorithms/101.js
let isSymmetric = function(root) {
  let left = root.left, right = root.right;
  if (!left && !right) return true;
  if (left == null) return false;
  if (right == null) return false;

  // Use a stack for each subtree.
  const sleft = [left], sright = [right];
  while (sleft.length || sright.length) {
    let [left, right] = [sleft.pop(), sright.pop()];
    // Value check.
    if (left.val != right.val) return false;

    // grab children for each
    let lleft = left.left, lright = left.right, rleft = right.left,
        rright = right.right;
    // and append them symmetrically.
    if (lleft && rright) {
      sleft.push(lleft);
      sright.push(rright);
    } else {
      // One of them is None. If not both,
      // we got a mismatch.
      if (lleft != rright) return false;
    }
    if (lright && rleft) {
      sleft.push(lright);
      sright.push(rleft);
    } else {
      if (lright != rleft) return false;
    }
  }
  return sleft.length == sright.length;
};